### PR TITLE
Add in-app light-box for viewing images.

### PR DESCRIPTION
**Pull request by Sam1301**
Fixes: #264

Added `LightBoxScreen`to 
* zoom-in/out 
* share
* download images

### CODE REVIEW

**Comment by Sam1301** [comment]
Made some additional layout changes (https://drive.google.com/open?id=0B4jK5QX65b0Zek8wWkR5UFllNEE).

File: src/light-box/LightBox.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
```
Comment (Sam1301) [inline src/light-box/LightBox.js]: Added an optional right button to ModalNavBar.

File: src/message/html/HtmlTagImg.js [hunk 2]
```diff
@@ -10,10 +13,32 @@ const styles = StyleSheet.create({
...
-    style={[styles.img, style]}
-  />
-);
+class HtmlTagImg extends Component {
+  constructor(props) {
+    super(props);
+    this.onPress = this.onPress.bind(this);
+
+    const { src, auth } = this.props;
+    this.fullUrl = getResource(src, auth);
```
Comment (Sam1301) [inline src/message/html/HtmlTagImg.js]: yeah, this was a typo, I had meant to add it to props. But using it directly in render certainly makes sense.

File: src/message/html/HtmlTagImg.js [hunk 3]
```diff
@@ -10,10 +13,32 @@ const styles = StyleSheet.create({
...
+  }
+
+  onPress() {
```
Comment (Sam1301) [inline src/message/html/HtmlTagImg.js]: understood 👍

**Comment by timabbott** [comment]
Mostly a sidenote, but @borisyankov does it make sense to switch the coveralls configuration to just do the "checks" / status update part, and not also the PR comments?  They're kinda spammy.

**Comment by borisyankov** [comment]
It absolutely does, @timabbott I think it was a form of learned helplessness to endure these messages when they don't make sense.

**Comment by borisyankov** [comment]
I think we want to get the React Navigation PR in, before merging this.
Let's try to have it soon to master, then resume work on this PR.

**Comment by Sam1301** [comment]
Updated! Now share, download and zoom works on both iOS and Android. Also reworked previous UI.

Android
screenshot_20170531-174123
screenshot_20170531-174200
screenshot_20170531-174223

iOS
screen shot 2017-05-31 at 6 30 56 pm
screen shot 2017-05-31 at 6 31 16 pm
screen shot 2017-05-31 at 6 31 29 pm

**Comment by borisyankov** [comment]
Cool. That covers a lot of things... which is also a concern.
We will want to split it in several PRs?

**Comment by Sam1301** [comment]
completely agree 👍

**Comment by Sam1301** [comment]
before splitting this, I am working on two issues the current version has
1) there is some internal error in react-native-fetch-blob which is causing the tests to fail. https://github.com/itinance/react-native-fs could be a better alternative (studying it).
2) adding some tests for light-box dir

**Comment by borisyankov** [comment]
Great. Keep it up then ;)

**Comment by borisyankov** [comment]
Is this PR still relevant? Anything left unmerged?

**Comment by Sam1301** [comment]
nope, everything is merged 👍

**Pull request closed without merging**